test/integration/scheduler/scheduler_test.go

```diff
@@ -338,4 +368,150 @@ var _ = ginkgo.Describe("Scheduler", func() {
+		// Simulated deletion of workload by ownerReference of job1.
+		gomega.Expect(k8sClient.Delete(ctx,
+			&kueue.QueuedWorkload{ObjectMeta: metav1.ObjectMeta{
+				Name:      job1.Name,
+				Namespace: job1.Namespace,
+			}})).Should(gomega.Succeed())
+
+		gomega.Eventually(func() *bool {
+			lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob2)).Should(gomega.Succeed())
+			return createdJob2.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
+		gomega.Expect(createdJob2.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+	})
+
+	ginkgo.It("Should re-enqueue by the updated event of ClusterQueue", func() {
+		job1 := testing.MakeJob("on-demand-job1", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "8").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
+
+		createdJob1 := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
+			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob1)).Should(gomega.Succeed())
+			return createdJob1.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
+		gomega.Expect(createdJob1.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+
+		job2 := testing.MakeJob("on-demand-job2", ns.Name).Queue(devBEQueue.
+			Name).Request(corev1.ResourceCPU, "8").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
+		createdJob2 := &batchv1.Job{}
+		gomega.Consistently(func() *bool {
+			lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob2)).Should(gomega.Succeed())
+			return createdJob2.Spec.Suspend
+		}, framework.ConsistentDuration, framework.Interval).Should(gomega.Equal(pointer.Bool(true)))
+
+		job3 := testing.MakeJob("on-demand-job3", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job3)).Should(gomega.Succeed())
+		createdJob3 := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
+			lookupKey := types.NamespacedName{Name: job3.Name, Namespace: job3.Namespace}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob3)).Should(gomega.Succeed())
+			return createdJob3.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
+		gomega.Expect(createdJob3.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+
+		ginkgo.By("updating ClusterQueue")
+		devCq := &kueue.ClusterQueue{}
+		gomega.Eventually(func() error {
+			lookupKey := types.NamespacedName{Name: devBEClusterQ.Name}
+			return k8sClient.Get(ctx, lookupKey, devCq)
+		}, framework.Timeout, framework.Interval).Should(gomega.Succeed())
```

Comment:
no need for the eventually
```suggestion
gomega.Expect(k8sClient.Get(ctx, lookupKey, devCq)).Should(gomega.Succeed())
```

Reply:
Done